src/sanitizer.js

```diff
@@ -333,38 +346,59 @@ export function sanitizeHtml(html) {
 }
 
 /**
- * Sanitizes the provided formatting HTML. Only the most basic inline tags are
- * allowed, such as <b>, <i>, etc.
+ * Sanitizes user provided HTML, input to mustache templates or the date picker.
  *
+ * @private
  * @param {string} html
  * @return {string}
  */
-export function sanitizeFormattingHtml(html) {
-  return htmlSanitizer.sanitizeWithPolicy(html,
-      function(tagName, attribs) {
-        if (tagName == 'template') {
-          for (let i = 0; i < attribs.length; i += 2) {
-            if (attribs[i] == 'type' && attribs[i + 1] == 'amp-mustache') {
-              return {
-                tagName,
-                attribs: ['type', 'amp-mustache'],
-              };
-            }
-          }
+export function sanitizeHtmlTags(html) {
+  /**
+   * Strip out the style attribute and its values.
+   * @param {!Array} attribs collection of attributes and values.
+   * @return {!Array}
+   */
+  const stripOutStyleAttributeAndValues = function(attribs) {
+    if (!isExperimentOn(self, 'inline-styles')) {
+      for (let i = 0; i < attribs.length; i += 2) {
+        if (attribs[i] == 'style') {
```

Comment:
having looked through the code, sanitizeFormattingHtml is the first pass for sanitizing html in mustache templates. sanitizeHtml, which is then called aftter, further sanitizes the html output from that output..stripping out unsafe event listeners and etc...And yes we do use sanitizeFormattingHtml exclusively in amp-date-picker. So both methods do similar things..but sanitizeFormattingHtml was meant to be less strict in its sanitizing logic. 

As discussed offline, i think it's better to be safe and keep the inline style logic in both places..though to think of it..maybe it makes even more sense to just place that particular sanitizer logic directly into the extensions where they are used...as opposed to it being in runtime..which you briefly touched on in our recent meeting. 

I will experiment with making those changes. But perhaps outside the scope of this PR

Reply:
Please take another look.

Changes:

 -  removed the usage of sanitizeFormattingHtml from amp-date-picker in #14673, as it was not in use. 
- renamed sanitizeHtmlTags to sanitizeTagsForTripleMustache as now it is specific to only amp-mustache.  Removed stripping of style attribute in that method, as I placed it there initially for defensive programming because i saw that it was used in amp-date-picker, and wanted to ensure that any usage of the method elsewhere would sanitize the style attribute. 
- added comments to ensure that people are aware of the lack of style sanitizing in sanitizeTagsForTripleMustache as there is currently no means to encapsulating that method specifically to amp mustache. Looking into how we can fix this..as this seems to be an issue with how sanitizer logic is used by the extensions amp-bind and amp-mustache.